test/jdk/sun/security/provider/acvp/Launcher.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import jdk.test.lib.json.JSONValue;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.Provider;
+import java.security.Security;
+
+/*
+ * @test
+ * @bug 8342442
+ * @library /test/lib
+ */
+public class Launcher {
+
+    private static final String ONLY_ALG
+            = System.getProperty("test.acvp.alg");
+
+    private static final Provider PROVIDER;
+
+    static {
+        var provProp = System.getProperty("test.acvp.provider");
+        PROVIDER = provProp != null
+                ? Security.getProvider(provProp)
+                : null;
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        // This test runs on "internalProjection.json"-style files generated
+        // by NIST's ACVP Server. See https://github.com/usnistgov/ACVP-Server.
+        //
+        // The files are either put into the "data" directory or another
+        // directory specified by the "test.acvp.data" test property.
+        // The test walks through the directory recursively and looks for
+        // file names equals to or ending with "internalProjection.json" and
+        // runs test on them.
+        //
+        // Set the "test.acvp.alg" test property to only test this algorithm.
+        //
+        // Sample files can be downloaded from
+        // https://github.com/usnistgov/ACVP-Server/tree/master/gen-val/json-files.
+        //
+        // By default, the test uses system-preferred implementations.
+        // If you want to test on a specific provider, set the
+        // "test.acvp.provider" test property. The provider must be
+        // registered.
+        //
+        // Tests for each algorithm must be compliant to its specification linked from
+        // https://github.com/usnistgov/ACVP?tab=readme-ov-file#supported-algorithms.
+        //
+        // Example:
+        //
+        // jtreg -Dtest.acvp.provider=SunJCE \
+        //       -Dtest.acvp.alg=ML-KEM \
+        //       -Dtest.acvp.data=/path/to/json-files/ \
+        //       -jdk:/path/to/jdk Launcher.java
```

Comment:
This seems more like either the class javadoc or the javadoc for main().  (Not an inline comment).

Reply:
Moved to class level.